domain/src/test/java/org/oppia/android/domain/oppialogger/analytics/FeatureFlagsLoggerTest.kt

```diff
@@ -0,0 +1,176 @@
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.app.model.EventLog.FeatureFlagContext.FeatureFlagItem
+import org.oppia.android.app.model.PlatformParameter.SyncStatus
+import org.oppia.android.domain.oppialogger.EventLogStorageCacheSize
+import org.oppia.android.domain.oppialogger.ExceptionLogStorageCacheSize
+import org.oppia.android.domain.oppialogger.LoggingIdentifierModule
+import org.oppia.android.domain.platformparameter.PlatformParameterSingletonModule
+import org.oppia.android.testing.FakeAnalyticsEventLogger
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.testing.logging.EventLogSubject.Companion.assertThat
+import org.oppia.android.testing.logging.SyncStatusTestModule
+import org.oppia.android.testing.platformparameter.TestPlatformParameterModule
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestCoroutineDispatchers
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.caching.AssetModule
+import org.oppia.android.util.data.DataProvidersInjector
+import org.oppia.android.util.data.DataProvidersInjectorProvider
+import org.oppia.android.util.locale.LocaleProdModule
+import org.oppia.android.util.logging.EnableConsoleLog
+import org.oppia.android.util.logging.EnableFileLog
+import org.oppia.android.util.logging.GlobalLogLevel
+import org.oppia.android.util.logging.LogLevel
+import org.oppia.android.util.networking.NetworkConnectionUtilDebugModule
+import org.oppia.android.util.platformparameter.DOWNLOADS_SUPPORT
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [FeatureFlagsLogger]. */
+// FunctionName: test names are conventionally named with underscores.
+@Suppress("FunctionName")
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = FeatureFlagsLoggerTest.TestApplication::class)
+class FeatureFlagsLoggerTest {
+  private companion object {
+    private const val TEST_SESSION_ID = "test_session_id"
+    private val INITIAL_SYNC_STATUS = SyncStatus.NOT_SYNCED_FROM_SERVER
+    private val testEnableDownloadsSupportFeatureFlagItem = FeatureFlagItem.newBuilder()
+      .setFlagName(DOWNLOADS_SUPPORT)
+      .setFlagEnabledState(false)
+      .setFlagSyncStatus(INITIAL_SYNC_STATUS)
+      .build()
+  }
+
+  @Inject lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+  @Inject lateinit var featureFlagsLogger: FeatureFlagsLogger
+  @Inject lateinit var fakeAnalyticsEventLogger: FakeAnalyticsEventLogger
+
+  @Before
+  fun setup() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testLogFeatureFlags_logsFeatureFlagsWithCorrectSessionId() {
```

Comment:
Test naming convention is `testAction_withOneCondition_withSecondCondition_hasExpectedOutcome` per [wiki](https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing#naming-convention).
 
```suggestion
  fun testLogFeatureFlags_logFeatureFlags_hasCorrectSessionId() {
```

Reply:
I have added this suggestion.